resolver.go

```diff
@@ -233,15 +235,27 @@ func (r *resolver) ServeDNS(w dns.ResponseWriter, query *dns.Msg) {
 		for i := 0; i < num; i++ {
 			log.Debugf("Querying ext dns %s:%s for %s[%d]", w.LocalAddr().Network(), r.extDNS[i], name, query.Question[0].Qtype)
 
-			c := &dns.Client{Net: w.LocalAddr().Network()}
-			addr := fmt.Sprintf("%s:%d", r.extDNS[i], 53)
+			var extConn net.Conn
+			r.sb.execFunc(func() {
```

Comment:
@mrjana Yes, creating the connection during the query is sub-optimal. First I tried creating the connection as part of the SetupFunc. But the Dial fails; probably because the external connectivity is not fully setup yet.. The error I was getting was "dial udp 8.8.8.8:53: connect: network is unreachable".

Can we do this instead; on a first external query lets create the connection and let it persist after that ?